add_to_app/android_view/app/src/main/java/dev/flutter/example/androidView/FlutterViewEngine.kt

```diff
@@ -0,0 +1,151 @@
+ *  <li>Splash screens. You must implement it yourself. Read from addOnFirstFrameRenderedListener
+ *      as needed.
+ *  <li>Transparency, surface/texture. These are just FlutterView level APIs. Set them on the
+ *      FlutterView directly.
+ *  <li>Intents. This doesn't do any translation of intents into actions in the FlutterEngine.
+ *      you must do them yourself.
+ *  <li>Back buttons. You must decide whether to send it to Flutter via
+ *      FlutterEngine.getNavigationChannel.popRoute(), or consume it natively. Though that decision
+ *      may be difficult due to https://github.com/flutter/flutter/issues/67011.
+ *  </ul>
+ *
+ * Your own FlutterView integrating application may need a similar wrapper but
+ * you must decide when the FlutterView is active based on your own application's
+ * intent.
+ */
+class FlutterViewEngine(val engine: FlutterEngine) : LifecycleObserver{
+    private var flutterView: FlutterView? = null
+    private var activity: ComponentActivity? = null
+    private var platformPlugin: PlatformPlugin? = null
+
+    // This is the intersection of an available activity and of a visible
+    // FlutterView. This is where Flutter would start rendering.
+    private fun hookActivityAndView() {
+        // Assert state.
+        activity!!.let { activity ->
+            flutterView!!.let { flutterView ->
+                platformPlugin = PlatformPlugin(activity, engine.platformChannel)
+
+                engine.activityControlSurface.attachToActivity(activity, activity.lifecycle)
+                flutterView.attachToFlutterEngine(engine)
+                activity.lifecycle.addObserver(this)
+            }
+        }
+    }
+
+    // Lost the intersection of either an available activity or a visible
+    // FlutterView.
+    private fun unhookActivityAndView() {
+        // Stop reacting to activity events.
+        activity!!.lifecycle.removeObserver(this)
+
+        // Plugins are no longer attached to an activity.
+        engine.activityControlSurface.detachFromActivity()
+
+        // Release Flutter's control of UI such as system chrome.
+        platformPlugin!!.destroy()
+        platformPlugin = null
+
+        // Set Flutter's application state to detached.
+        engine.lifecycleChannel.appIsDetached();
+
+        // Detach rendering pipeline.
+        flutterView!!.detachFromFlutterEngine()
+    }
+
+    fun attachToActivity(activity: ComponentActivity) {
+        this.activity = activity
+        if (flutterView != null) {
+            hookActivityAndView()
+        }
+    }
+
+    fun detachActivity() {
+        if (flutterView != null) {
+            unhookActivityAndView()
+        }
+        activity = null
+    }
+
+    fun attachFlutterView(flutterView: FlutterView) {
+        this.flutterView = flutterView
+        if (activity != null) {
+            hookActivityAndView()
+        }
+    }
+
+    fun detachFlutterView() {
```

Comment:
For sample code should we have full docstring documentation?

Reply:
done